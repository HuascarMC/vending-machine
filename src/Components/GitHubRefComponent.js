import React from 'react';

class GitHubRefComponent extends React.Component {
  render() {
    return(
      <div className="github-logo">
        <a href='https://github.com/HuascarMC/vending-machine-server/tree/heroku-deploy' target="_blank"><img src='/images/github-logo.png'/></a>
      </div>
    );
  }
}


export default GitHubRefComponent;
