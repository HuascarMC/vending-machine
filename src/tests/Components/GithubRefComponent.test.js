import React from 'react';
import GitHubRefComponent from '../../Components/GitHubRefComponent'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('should render without crashing', () => {
  shallow(<GitHubRefComponent />);
});

it('should render github-logo div', () => {
  const wrapper = shallow(<GitHubRefComponent/>);
  const div = wrapper.find('.github-logo');
  expect(div.exists()).toEqual(true);
});

it('should render hyperlink tag', () => {
  const wrapper = shallow(<GitHubRefComponent/>);
  const a = wrapper.find('a');
  expect(a.exists()).toEqual(true);
})
