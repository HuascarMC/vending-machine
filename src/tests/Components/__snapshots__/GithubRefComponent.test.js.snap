// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="github-logo"
>
  <a
    href="https://github.com/HuascarMC/vending-machine-server/tree/heroku-deploy"
    target="_blank"
  >
    <img
      src="/images/github-logo.png"
    />
  </a>
</div>
`;
